superset/daos/base.py

```diff
@@ -171,10 +171,17 @@ def update(cls, model: T, properties: dict[str, Any], commit: bool = True) -> T:
         Generic update a model
         :raises: DAOCreateFailedError
         """
+        d = dict(model.__dict__)
+        d.pop("_sa_instance_state")  # get rid of SQLAlchemy special attr
+        copy = model.__class__(**d)
+        copy.id = model.id
+        # create a copy of the model here
+        # so it doesn't trigger before_update/after_update sqla listener each time
+        # we set a property
```

Comment:
this is an important change to the base update method.
By creating a out of session copy of the original model instance, we can set attributes on the copy without worrying the before/after update sqla listener will be triggered for each property.

The listener will only be triggered when session is committed

Reply:
there's one test `test_run_sync_query_dont_exist` that I cannot get passed by with this baseDao.update method change(using a copy model instance out of session and merge)
We probably need more thoughts on how to reduce number of events triggered on each property update, thus reverting here